### PR TITLE
Extendable selector

### PR DESCRIPTION
**Pull request by AronBuzogany**
This PR proposes a way to display the description of a query defined in the configuration file. 

By clicking on the little arrow next to the query name, the description will be displayed under the selection field.

### CODE REVIEW

**Comment by pheyvaer** [comment]
Executing the queries gives no results, but it does give empty rows:

image

Also the column are not removed from previous queries.

**Comment by AronBuzogany** [comment]
> Initially executing the queries gives no results, but it does give empty rows:
> 
> image
> 
> When I click on "Refresh" data is added though.
> 
> Also the columns are not removed from previous queries:
> 
> image

This issue is now resolved

**Comment by pheyvaer** [comment]
I will review once the other PR is in `main` and `main` is merged in the branch of this PR.

File: src/selection-table/SelectionTable.css [hunk 1]
```diff
@@ -30,6 +30,59 @@
...
+    border: .5em solid transparent;
+    border-left-color: gray;
+    transform-origin: 0 50%;
+    transition: transform .25s;
+  }
+  
+  .arrow.down::before {
+    transform: rotate(90deg);
+    transition: transform .25s;
+  }
```
Comment (pheyvaer) [inline src/selection-table/SelectionTable.css:79]: Fix formatting.

File: web-app/src/selection-table/SelectionTable.css [hunk 2]
```diff
@@ -0,0 +1,87 @@
+
```
Comment (pheyvaer) [inline web-app/src/selection-table/SelectionTable.css:1]: Is this file not a duplicate from `src/selection-table/SelectionTable.css`?

Comment (pheyvaer) [reply web-app/src/selection-table/SelectionTable.css:1]: And `web-app/src/selection-table/SelectionTable.jsx` is empty.

Comment (AronBuzogany) [reply web-app/src/selection-table/SelectionTable.css]: I am assuming this was a merge mistake, I have completely removed the web-app folder in [commit 781880c](https://github.com/SolidLabResearch/generic-data-viewer/pull/3/commits/781880c1857d597a9fd70af62c8a3756188adcb2)

File: src/selection-table/SelectionTable.css (hunk 1 above)
Comment (AronBuzogany) [reply src/selection-table/SelectionTable.css]: This, among other things is resolved in [commit a812905](https://github.com/SolidLabResearch/generic-data-viewer/pull/3/commits/a8129055504bbe5471cbfdfe3a7b882e857b70fd)

File: src/components/ExtendableSelector.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+function ExtendableSelector({onClick, extendContent, children}){
+    const [extended, setExtended] = useState(false)
+    let switchExtended = () => setExtended((old) => {return !old})
+    return(
+        <>
+            <li onClick={onClick} className="query-entry">
+                <div className="extend-container">
+                     {children}
+                     <div onClick={(e) => {switchExtended(); e.target.classList.toggle('down')}} className="arrow"/>
```
[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/components/ExtendableSelector.jsx]: I have changed it to be a button in [commit a812905](https://github.com/SolidLabResearch/generic-data-viewer/pull/3/commits/a8129055504bbe5471cbfdfe3a7b882e857b70fd)

[the remark this answers is not in the document]
Comment (AronBuzogany) [reply src/components/ExtendableSelector.jsx]: The arrow extending the query-selectors is also clickable, I have adjusted it respectively in [commit b87cdaa](https://github.com/SolidLabResearch/generic-data-viewer/pull/3/commits/b87cdaaee4d19bcb3ad174f7c4efc0a3ba1a6ff9)

**Review by pheyvaer — APPROVED** [review]

**Pull request merged**